locations/spiders/wyndham.py

```diff
@@ -0,0 +1,187 @@
+    "Accept": "*/*",
+    "Accept-Language": "en-US,en;q=0.5",
+    "Accept-Encoding": "gzip, deflate, br",
+    "X-Requested-With": "XMLHttpRequest",
+    "DNT": "1",
+    "Connection": "keep-alive",
+    "Referer": "https://www.wyndhamhotels.com/locations",
+    "TE": "Trailers",
+}
+
+
+
+def create_url(brand, city, state, unique_url, tier_id):
+    # In the html, there is a script which this recreates
+    brand_name = BRAND_MAP.get(brand)
+    if not brand_name:
+        brand_name = BRAND_MAP.get(tier_id.lower())
+    url = brand_name + "/"
+    url += city.replace(" ", "-").lower()
+    state_name = (
+        f"-{state.replace(' ','-').lower()}"
+        if state.lower() != "other than us/canada"
+        else ""
+    )
+    url += state_name + "/"
+    url += unique_url.lower() + "/"
+    url += "overview"
+    return url
+
+
+class WyndhamSpider(scrapy.Spider):
+    name = "wyndham"
+    allowed_domains = ["www.wyndhamhotels.com"]
+    headers = HEADERS
+
+    def start_requests(self):
+        start_url = "https://www.wyndhamhotels.com/BWSServices/services/search/properties?recordsPerPage=&pageNumber=1&brandId=ALL&countryCode=&noPropertyData=false"
+        yield scrapy.Request(start_url, callback=self.parse, headers=self.headers)
+
+    def parse(self, response):
+        data = json.loads(response.text)
+        for country in data["countries"]:
+            country_code = country["countryCode"]
+            for state in country["states"]:
+                state_name = state["stateName"]
+                for city in state["cities"]:
+                    city_name = city["cityName"]
+                    for property_ in city["propertyList"]:
+                        property_id = property_["propertyId"]
+                        brand_id = property_["brandId"]
+                        brand_name = property_["brand"]
+                        brand_tier = property_["tierId"]
+                        unique_url = property_["uniqueUrl"]
+                        url = create_url(
+                            brand_id, city_name, state_name, unique_url, brand_tier
+                        )
+                        yield scrapy.Request(
+                            f"https://{self.allowed_domains[0]}/{url}",
+                            self.parse_property,
+                            meta={
+                                "id": property_id,
+                                "country_code": country_code,
+                                "brand_name": brand_name,
+                            },
+                            headers=HEADERS,
+                        )
+
+    def parse_property(self, response):
+        raw_json = re.search(
+            '<script type="application\/ld\+json"\>(.+?)\<',
+            response.text,
+            flags=re.DOTALL,
+        )
+        if not raw_json:
+            return None
+        data = json.loads(raw_json.group(1))
```

Comment:
[The logs](https://placescraper-results.s3.amazonaws.com/ci/319483762/wyndham/log.txt) show that the json parse fails on at least one page. That's why the CI fails, but it'll work in the weekly scrapes.

Reply:
Didn't like the /t character; updated to strip those